ghost/members-ssr/lib/members-ssr.js

```diff
@@ -201,43 +201,51 @@ class MembersSSR {
-        if (!req.url) {
-            return Promise.reject(new BadRequestError({
-                message: 'Expected token param containing JWT'
-            }));
-        }
+        try {
+            if (!req.url) {
+                return Promise.reject(new BadRequestError({
+                    message: 'Expected token param containing JWT'
+                }));
+            }
 
-        const {query} = parseUrl(req.url, true);
-        if (!query || !query.token) {
-            return Promise.reject(new BadRequestError({
-                message: 'Expected token param containing JWT'
-            }));
-        }
+            const {query} = parseUrl(req.url, true);
+            if (!query || !query.token) {
+                return Promise.reject(new BadRequestError({
+                    message: 'Expected token param containing JWT'
+                }));
+            }
 
-        const token = Array.isArray(query.token) ? query.token[0] : query.token;
-        const member = await this._getMemberDataFromToken(token);
+            const token = Array.isArray(query.token) ? query.token[0] : query.token;
+            const member = await this._getMemberDataFromToken(token);
 
-        if (!member) {
-            // The member doesn't exist any longer (could be a sign in token for a member that was deleted)
-            return Promise.reject(new BadRequestError({
-                message: 'Invalid token'
-            }));
-        }
+            if (!member) {
+                // The member doesn't exist any longer (could be a sign in token for a member that was deleted)
+                return Promise.reject(new BadRequestError({
+                    message: 'Invalid token'
+                }));
+            }
 
-        // perform and store geoip lookup for members when they log in
-        if (!member.geolocation) {
-            try {
-                await this._setMemberGeolocationFromIp(member.email, req.ip);
-            } catch (err) {
-                // no-op, we don't want to stop anything working due to
-                // geolocation lookup failing
-                debug(`Geolocation lookup failed: ${err.message}`);
+            // perform and store geoip lookup for members when they log in
+            if (!member.geolocation) {
+                try {
+                    await this._setMemberGeolocationFromIp(member.email, req.ip);
+                } catch (err) {
+                    // no-op, we don't want to stop anything working due to
+                    // geolocation lookup failing
+                    debug(`Geolocation lookup failed: ${err.message}`);
+                }
             }
-        }
 
-        this._setSessionCookie(req, res, member.email);
+            this._setSessionCookie(req, res, member.email);
 
-        return member;
+            return member;
+        } catch (err) {
+            if (err.errorType === 'BadRequestError') {
+                return Promise.reject(new BadRequestError({
```

Comment:
This is handled [here](https://github.com/TryGhost/Ghost/blob/637f5c927c292f30852763914e22cfa2c715f28f/ghost/core/core/server/services/members/middleware.js#L285).
The behavior is that, when an invalid id is passed, do a standard 302 redirect to the homepage, with success=false. 
This is different from the 400 error we discussed in the tech spec but this felt more standard depending on how we are handling `BadRequestError` currently. 
What are your thoughts? Does this approach look fine?

<img width="1727" alt="image" src="https://github.com/TryGhost/Ghost/assets/17474532/6ba8e4be-89db-46c8-9c32-02722610b482">

    
<img width="1060" alt="image" src="https://github.com/TryGhost/Ghost/assets/17474532/d4fcb555-7174-4e56-8891-e802d2481929">

Reply:
I'm not sure Im following - Why do we need to make a change to `exchangeTokenForSession` for the work we are doing? I would not have expected to need to touch this area of the codebase... The diff is a bit hard to grok, but it looks like you have just wrapped the contents of the function in a try..catch?